Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -92,6 +101,356 @@ namespace ScriptAutomation
+
+        AZStd::string GetScreenshotOutputPath(bool normalized)
+        {
+            return Utils::GetScreenshotsPath(normalized);
+        }
+
+        AZStd::string GetProfilingOutputPath(bool normalized)
+        {
+            return Utils::GetProfilingPath(normalized);
+        }
+
+        bool PrepareForScreenCapture(const AZStd::string& path)
+        {
+            if (!Utils::IsFileUnderFolder(Utils::ResolvePath(path), Utils::GetScreenshotsPath(true)))
+            {
+                // The main reason we require screenshots to be in a specific folder is to ensure we don't delete or replace some other important file.
+                AZ_Error("ScriptAutomation", false, "Screenshots must be captured under the '%s' folder. Attempted to save screenshot to '%s'.",
+                    Utils::GetScreenshotsPath(false).c_str(), path.c_str());
+
+                return false;
+            }
+
+            // Delete the file if it already exists
+            if (AZ::IO::LocalFileIO::GetInstance()->Exists(path.c_str()) && !AZ::IO::LocalFileIO::GetInstance()->Remove(path.c_str()))
+            {
+                AZ_Error("ScriptAutomation", false, "Failed to delete existing screenshot file '%s'.", path.c_str());
+                return false;
+            }
+
+            auto ScriptAutomationInterface = ScriptAutomationInterface::Get();
+            auto automationComponent = azrtti_cast<ScriptAutomationSystemComponent*>(ScriptAutomationInterface);
+
+            automationComponent->SetCapturePending(true);
+            ScriptAutomationInterface->PauseAutomation();
+
+            return true;
+        }
+
+        void CaptureScreenshot(const AZStd::string& filePath)
+        {
+            auto operation = [filePath]()
+            {
+                // Note this will pause the script until the capture is complete
+                if (PrepareForScreenCapture(filePath))
+                {
+                    auto ScriptAutomationInterface = ScriptAutomationInterface::Get();
+                    auto automationComponent = azrtti_cast<ScriptAutomationSystemComponent*>(ScriptAutomationInterface);
+                    uint32_t frameCaptureId = AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId;
+                    AZ::Render::FrameCaptureRequestBus::BroadcastResult(frameCaptureId, &AZ::Render::FrameCaptureRequestBus::Events::CaptureScreenshot, filePath);
+                    if (frameCaptureId != AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId)
+                    {
+                        automationComponent->SetFrameCaptureId(frameCaptureId);
+                        automationComponent->AZ::Render::FrameCaptureNotificationBus::Handler::BusConnect();
+                    }
+                    else
+                    {
+                        AZ_Error("ScriptAutomation", false, "Script: failed to trigger screenshot capture");
+                       ScriptAutomationInterface->ResumeAutomation();
+                    }
```

Comment:
This is a bit of a nitpick, but there's a bit of inconsistency here.
For the pausing, the approach is to take some action in PrepareForScreenCapture and then reverse the action here if there's an error.
For FrameCaptureNotificationBus, the approach is to wait until we confirm firm CaptureScreenshot result, and then connect to the bus.

To be consistent, I would connect the FrameCaptureNotificationBus in PrepareForScreenCapture, and then disconnect the bus here if there's an error.

Reply:
The FrameCaptureNotification bus is because I know there is a change coming where the frame capture id is used as the notification bus address, so you can't connect until after the capture is triggered.

Reply:
To answer your original question, I do have the equivalent change local for ASV to go along with my change to the FrameCaptureNotificationBus. I will put that up for review when I get closer the FCNB change being committed.